### PR TITLE
java CDK, source-postgres: speed up tests involving debezium

### LINKED ISSUE #31828: java CDK: make DebeziumRecordIterator less time-sensitive in tests
### Topic

_No response_

### Relevant information

We decide that the DebeziumEngine is not going to be sending us any more records by waiting a certain amount of time and then giving up.

For tests, at least, this isn't desirable. Shouldn't we be able to decide that the WAL isn't going to have anything interesting in it using some LSN offset arithmetic? Something needs to be done here. This is a major cause of latency in the source-postgres CI: https://github.com/airbytehq/airbyte/issues/31298#issuecomment-1779261738

### PR DESCRIPTION
**Pull request by postamar**
This PR includes three meaningful, distinct changes which each improve test execution latency when relying on the Debezium engine. These changes aim to have limited impact in a production setting.

Fixes https://github.com/airbytehq/airbyte/issues/31828.

### CODE REVIEW

File: airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/cdc/PostgresCdcCtidInitializer.java [hunk 1]
```diff
@@ -167,6 +167,8 @@ public static List<AutoCloseableIterator<AirbyteMessage>> cdcCtidIteratorsCombin
       final PostgresCdcStateHandler postgresCdcStateHandler = new PostgresCdcStateHandler(stateManager);
 
       final boolean canShortCircuitDebeziumEngine = savedOffset.isPresent() &&
+      // Until the need presents itself in production, short-circuiting should only be done in tests.
+          sourceConfig.has("is_test") && sourceConfig.get("is_test").asBoolean() &&
```
Comment (rodireich) [inline airbyte-integrations/connectors/source-postgres/src/main/java/io/airbyte/integrations/source/postgres/cdc/PostgresCdcCtidInitializer.java:171]: Thank you

**Review by rodireich — APPROVED** [review]

**Comment by postamar** [comment]
Hmm, seems like CDK version 0.2.1 has already been published. Bumping to 0.2.2.

**Comment by postamar** [comment]
Thanks for the review!

**Pull request merged**